keyboards/keychron/k12/keymaps/via/k12_via_ansi.json

```
@@ -0,0 +1,315 @@
+{
+  "name": "Keychron K12 ANSI",
+  "vendorId": "0x3434",
+  "productId": "0xFE16",
+  "lighting": {
+    "extends": "qmk_rgblight",
+    "underglowEffects": [
+      [
+        "None",
+        0
+      ],
+      [
+        "SOLID_COLOR",
+        1
+      ],
+      [
+        "ALPHAS_MODS",
+        1
+      ],
+      [
+        "GRADIENT_UP_DOWN",
+        1
+      ],
+      [
+        "GRADIENT_LEFT_RIGHT",
+        1
+      ],
+      [
+        "BREATHING",
+        1
+      ],
+      [
+        "BAND_SAT",
+        1
+      ],
+      [
+        "BAND_VAL",
+        1
+      ],
+      [
+        "BAND_PINWHEEL_SAT",
+        1
+      ],
+      [
+        "BAND_PINWHEEL_VAL",
+        1
+      ],
+      [
+        "BAND_SPIRAL_SAT",
+        1
+      ],
+      [
+        "BAND_SPIRAL_VAL",
+        1
+      ],
+      [
+        "CYCLE_ALL",
+        1
+      ],
+      [
+        "CYCLE_LEFT_RIGHT",
+        1
+      ],
+      [
+        "CYCLE_UP_DOWN",
+        1
+      ],
+      [
+        "CYCLE_OUT_IN",
+        1
+      ],
+      [
+        "CYCLE_OUT_IN_DUAL",
+        1
+      ],
+      [
+        "RAINBOW_MOVING_CHEVRON",
+        1
+      ],
+      [
+        "CYCLE_PINWHEEL",
+        1
+      ],
+      [
+        "CYCLE_SPIRAL",
+        1
+      ],
+      [
+        "DUAL_BEACON",
+        1
+      ],
+      [
+        "RAINBOW_BEACON",
+        1
+      ],
+      [
+        "RAINBOW_PINWHEELS",
+        1
+      ],
+      [
+        "RAINDROPS",
+        1
+      ],
+      [
+        "JELLYBEAN_RAINDROPS",
+        1
+      ],
+      [
+        "HUE_BREATHING",
+        1
+      ],
+      [
+        "HUE_PENDULUM",
+        1
+      ],
+      [
+        "HUE_WAVE",
+        1
+      ],
+      [
+        "PIXEL_FRACTAL",
+        1
+      ],
+      [
+        "PIXEL_RAIN",
+        1
+      ],
+      [
+        "TYPING_HEATMAP",
+        1
+      ],
+      [
+        "DIGITAL_RAIN",
+        1
+      ],
+      [
+        "SOLID_REACTIVE_SIMPLE",
+        1
+      ],
+      [
+        "SOLID_REACTIVE",
+        1
+      ],
+      [
+        "SOLID_REACTIVE_WIDE",
+        1
+      ],
+      [
+        "SOLID_REACTIVE_MULTIWIDE",
+        1
+      ],
+      [
+        "SOLID_REACTIVE_CROSS",
+        1
+      ],
+      [
+        "SOLID_REACTIVE_MULTICROSS",
+        1
+      ],
+      [
+        "SOLID_REACTIVE_NEXUS",
+        1
+      ],
+      [
+        "SOLID_REACTIVE_MULTINEXUS",
+        1
+      ],
+      [
+        "SPLASH",
+        1
+      ],
+      [
+        "MULTISPLASH",
+        1
+      ],
+      [
+        "SOLID_SPLASH",
+        1
+      ],
+      [
+        "SOLID_MULTISPLASH",
+        1
+      ]
+    ],
+    "supportedLightingValues": [
+      128,
+      129,
+      131
+    ]
+  },
```

for VIA V3
```suggestion
  "keycodes": ["qmk_lighting"],
  "menus": ["qmk_rgb_matrix"],
```
looks like you have all the animations enabled, so you don't need to list individual effects. If you want to, check VIA docs, v3 structure ( it's been moved)